x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/formatter/TextFormat.java

```diff
@@ -291,7 +291,7 @@ public Iterator<CheckedConsumer<Writer, IOException>> format(RestRequest request
         boolean[] dropColumns = dropNullColumns ? esqlResponse.nullColumns() : new boolean[esqlResponse.columns().size()];
         return Iterators.concat(
             // if the header is requested return the info
-            hasHeader(request) && esqlResponse.columns().isEmpty() == false
+            hasHeader(request) && esqlResponse.columns() != null
```

Comment:
I believe columns will never be null. However, changing the check to see if they are empty might cause some test cases to fail. Perhaps it should be updated to handle empty values and adjust the affected test cases accordingly.

Reply:
I didn't try it out, but it looks to me like it should produce the same result. But yes, seems like `esqlResponse.columns()` will currently never be `null`.